source/extensions/filters/common/ext_authz/ext_authz_http_impl.cc

```diff
@@ -310,22 +314,47 @@ ResponsePtr RawHttpClientImpl::toResponse(Http::ResponseMessagePtr message) {
     return std::make_unique<Response>(errorResponse());
   }
 
+  // Extract headers-to-remove from the storage header coming from the
+  // authorization server.
+  const auto& storage_header_name = Http::Headers::get().EnvoyAuthHeadersToRemove;
+  // If we are going to construct an Ok response we need to save the
+  // headers_to_remove in a variable first.
+  std::vector<Http::LowerCaseString> headers_to_remove{};
+  if (status_code == enumToInt(Http::Code::OK)) {
+    const Http::HeaderEntry* entry = message->headers().get(storage_header_name);
+    if (entry != nullptr) {
+      absl::string_view storage_header_value = entry->value().getStringView();
+      for (const auto header_name :
+           StringUtil::splitToken(storage_header_value, ",", false, true)) {
```

Comment:
nit: 

```c++
StringUtil::splitToken(storage_header_value, ",", /*keep_empty_string=*/false, /*trim_whitespace=*/true)
```

Reply:
And it seems like before pushing the headers, we can allocate (`reserve`) the size of `headers_to_remove` here?

Reply:
Fixing both of these.